### ISSUE TITLE
Cannot install package with pip on Ubuntu or MacOS

### ISSUE THREAD

**Issue by ragnarlonn** [loadimpact/loadimpact-cli#13]
1. First I tried "pip install loadimpact-cli" according to instructions in README.md:

```
Ragnars-MacBook-Air:~ ragnarlonn$ pip install loadimpact-cli
Collecting loadimpact-cli
  Downloading loadimpact-cli-1.0.6.tar.gz
Collecting setuptools>=18 (from loadimpact-cli)
  Downloading setuptools-34.0.2-py2.py3-none-any.whl (387kB)
    100% |████████████████████████████████| 389kB 786kB/s 
Collecting click (from loadimpact-cli)
  Downloading click-6.7-py2.py3-none-any.whl (71kB)
    100% |████████████████████████████████| 71kB 4.8MB/s 
Collecting loadimpact-v3 (from loadimpact-cli)
  Downloading loadimpact-v3-0.0.3.tar.gz
Collecting tzlocal (from loadimpact-cli)
  Downloading tzlocal-1.3.tar.gz
Requirement already satisfied (use --upgrade to upgrade): six in /System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python (from loadimpact-cli)
Collecting mock (from loadimpact-cli)
  Downloading mock-2.0.0-py2.py3-none-any.whl (56kB)
    100% |████████████████████████████████| 61kB 1.7MB/s 
Collecting appdirs>=1.4.0 (from setuptools>=18->loadimpact-cli)
  Downloading appdirs-1.4.0-py2.py3-none-any.whl
Collecting packaging>=16.8 (from setuptools>=18->loadimpact-cli)
  Downloading packaging-16.8-py2.py3-none-any.whl
Collecting requests (from loadimpact-v3->loadimpact-cli)
  Downloading requests-2.13.0-py2.py3-none-any.whl (584kB)
    100% |████████████████████████████████| 593kB 1.2MB/s 
Requirement already satisfied (use --upgrade to upgrade): pytz in /System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python (from tzlocal->loadimpact-cli)
Collecting funcsigs>=1; python_version < "3.3" (from mock->loadimpact-cli)
  Downloading funcsigs-1.0.2-py2.py3-none-any.whl
Collecting pbr>=0.11 (from mock->loadimpact-cli)
  Downloading pbr-1.10.0-py2.py3-none-any.whl (96kB)
    100% |████████████████████████████████| 102kB 2.2MB/s 
Requirement already satisfied (use --upgrade to upgrade): pyparsing in /System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python (from packaging>=16.8->setuptools>=18->loadimpact-cli)
Building wheels for collected packages: loadimpact-cli, loadimpact-v3, tzlocal
  Running setup.py bdist_wheel for loadimpact-cli ... done
  Stored in directory: /Users/ragnarlonn/Library/Caches/pip/wheels/1b/08/b7/b398baf90e3c8065103e89106c2042b1941b5316dc1d90e156
  Running setup.py bdist_wheel for loadimpact-v3 ... done
  Stored in directory: /Users/ragnarlonn/Library/Caches/pip/wheels/9d/d7/e6/5ae811e3067ad65219b6e8a5f568aad0377e57bdeb64d9c1e2
  Running setup.py bdist_wheel for tzlocal ... done
  Stored in directory: /Users/ragnarlonn/Library/Caches/pip/wheels/80/19/a8/635ad9f4ad8a63b49d073c55cbca31fb5898ce2560ed145a69
Successfully built loadimpact-cli loadimpact-v3 tzlocal
Installing collected packages: appdirs, packaging, setuptools, click, requests, loadimpact-v3, tzlocal, funcsigs, pbr, mock, loadimpact-cli
Exception:
Traceback (most recent call last):
  File "/Library/Python/2.7/site-packages/pip/basecommand.py", line 215, in main
    status = self.run(options, args)
  File "/Library/Python/2.7/site-packages/pip/commands/install.py", line 317, in run
    prefix=options.prefix_path,
  File "/Library/Python/2.7/site-packages/pip/req/req_set.py", line 742, in install
    **kwargs
  File "/Library/Python/2.7/site-packages/pip/req/req_install.py", line 831, in install
    self.move_wheel_files(self.source_dir, root=root, prefix=prefix)
  File "/Library/Python/2.7/site-packages/pip/req/req_install.py", line 1032, in move_wheel_files
    isolated=self.isolated,
  File "/Library/Python/2.7/site-packages/pip/wheel.py", line 346, in move_wheel_files
    clobber(source, lib_dir, True)
  File "/Library/Python/2.7/site-packages/pip/wheel.py", line 324, in clobber
    shutil.copyfile(srcfile, destfile)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/shutil.py", line 83, in copyfile
    with open(dst, 'wb') as fdst:
IOError: [Errno 13] Permission denied: '/Library/Python/2.7/site-packages/appdirs.py'
You are using pip version 8.1.2, however version 9.0.1 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
```

2. So, "permission denied" indicates lack of privileges to install. So I tried sudo:ing the command (README.md should maybe use "sudo" in the cmd line example)

```
Ragnars-MacBook-Air:~ ragnarlonn$ sudo pip install loadimpact-cli
Password:
The directory '/Users/ragnarlonn/Library/Caches/pip/http' or its parent directory is not owned by the current user and the cache has been disabled. Please check the permissions and owner of that directory. If executing pip with sudo, you may want sudo's -H flag.
The directory '/Users/ragnarlonn/Library/Caches/pip' or its parent directory is not owned by the current user and caching wheels has been disabled. check the permissions and owner of that directory. If executing pip with sudo, you may want sudo's -H flag.
Collecting loadimpact-cli
Collecting tzlocal (from loadimpact-cli)
Collecting loadimpact-v3 (from loadimpact-cli)
Requirement already satisfied (use --upgrade to upgrade): six in /System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python (from loadimpact-cli)
Collecting setuptools>=18 (from loadimpact-cli)
  Downloading setuptools-34.0.2-py2.py3-none-any.whl (387kB)
    100% |████████████████████████████████| 389kB 505kB/s 
Collecting mock (from loadimpact-cli)
  Downloading mock-2.0.0-py2.py3-none-any.whl (56kB)
    100% |████████████████████████████████| 61kB 2.1MB/s 
Collecting click (from loadimpact-cli)
  Downloading click-6.7-py2.py3-none-any.whl (71kB)
    100% |████████████████████████████████| 71kB 1.5MB/s 
Requirement already satisfied (use --upgrade to upgrade): pytz in /System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python (from tzlocal->loadimpact-cli)
Collecting requests (from loadimpact-v3->loadimpact-cli)
  Downloading requests-2.13.0-py2.py3-none-any.whl (584kB)
    100% |████████████████████████████████| 593kB 1.3MB/s 
Collecting appdirs>=1.4.0 (from setuptools>=18->loadimpact-cli)
  Downloading appdirs-1.4.0-py2.py3-none-any.whl
Collecting packaging>=16.8 (from setuptools>=18->loadimpact-cli)
  Downloading packaging-16.8-py2.py3-none-any.whl
Collecting funcsigs>=1; python_version < "3.3" (from mock->loadimpact-cli)
  Downloading funcsigs-1.0.2-py2.py3-none-any.whl
Collecting pbr>=0.11 (from mock->loadimpact-cli)
  Downloading pbr-1.10.0-py2.py3-none-any.whl (96kB)
    100% |████████████████████████████████| 102kB 1.6MB/s 
Requirement already satisfied (use --upgrade to upgrade): pyparsing in /System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python (from packaging>=16.8->setuptools>=18->loadimpact-cli)
Installing collected packages: tzlocal, requests, loadimpact-v3, appdirs, packaging, setuptools, funcsigs, pbr, mock, click, loadimpact-cli
  Found existing installation: setuptools 1.1.6
    Uninstalling setuptools-1.1.6:
Exception:
Traceback (most recent call last):
  File "/Library/Python/2.7/site-packages/pip/basecommand.py", line 215, in main
    status = self.run(options, args)
  File "/Library/Python/2.7/site-packages/pip/commands/install.py", line 317, in run
    prefix=options.prefix_path,
  File "/Library/Python/2.7/site-packages/pip/req/req_set.py", line 736, in install
    requirement.uninstall(auto_confirm=True)
  File "/Library/Python/2.7/site-packages/pip/req/req_install.py", line 742, in uninstall
    paths_to_remove.remove(auto_confirm)
  File "/Library/Python/2.7/site-packages/pip/req/req_uninstall.py", line 115, in remove
    renames(path, new_path)
  File "/Library/Python/2.7/site-packages/pip/utils/__init__.py", line 267, in renames
    shutil.move(old, new)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/shutil.py", line 299, in move
    copytree(src, real_dst, symlinks=True)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/shutil.py", line 208, in copytree
    raise Error, errors
Error: [('/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/_markerlib/__init__.py', '/tmp/pip-xlSv70-uninstall/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/_markerlib/__init__.py', "[Errno 1] Operation not permitted: '/tmp/pip-xlSv70-uninstall/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/_markerlib/__init__.py'"), ('/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/_markerlib/__init__.pyc', '/tmp/pip-xlSv70-uninstall/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/_markerlib/__init__.pyc', "[Errno 1] Operation not permitted: '/tmp/pip-xlSv70-uninstall/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/_markerlib/__init__.pyc'"), ('/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/_markerlib/markers.py', '/tmp/pip-xlSv70-uninstall/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/_markerlib/markers.py', "[Errno 1] Operation not permitted: '/tmp/pip-xlSv70-uninstall/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/_markerlib/markers.py'"), ('/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/_markerlib/markers.pyc', '/tmp/pip-xlSv70-uninstall/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/_markerlib/markers.pyc', "[Errno 1] Operation not permitted: '/tmp/pip-xlSv70-uninstall/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/_markerlib/markers.pyc'"), ('/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/_markerlib', '/tmp/pip-xlSv70-uninstall/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/_markerlib', "[Errno 1] Operation not permitted: '/tmp/pip-xlSv70-uninstall/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/_markerlib'")]
You are using pip version 8.1.2, however version 9.0.1 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
```

3. No idea what caused that error. I'll try cloning the repo and using setup.py instead:

[after cloning and CDing into the loadimpact-cli dir]

```
sudo python setup.py install
/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/distutils/dist.py:267: UserWarning: Unknown distribution option: 'test_requires'
  warnings.warn(msg)
running install
Checking .pth file support in /Library/Python/2.7/site-packages/
/usr/bin/python -E -c pass
TEST PASSED: /Library/Python/2.7/site-packages/ appears to support .pth files
running bdist_egg
running egg_info
creating loadimpact_cli.egg-info
writing requirements to loadimpact_cli.egg-info/requires.txt
writing loadimpact_cli.egg-info/PKG-INFO
writing top-level names to loadimpact_cli.egg-info/top_level.txt
writing dependency_links to loadimpact_cli.egg-info/dependency_links.txt
writing entry points to loadimpact_cli.egg-info/entry_points.txt
writing manifest file 'loadimpact_cli.egg-info/SOURCES.txt'
file loadimpactcli.py (for module loadimpactcli) not found
reading manifest file 'loadimpact_cli.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
warning: no previously-included files matching '__pycache__' found anywhere in distribution
writing manifest file 'loadimpact_cli.egg-info/SOURCES.txt'
installing library code to build/bdist.macosx-10.11-intel/egg
running install_lib
running build_py
file loadimpactcli.py (for module loadimpactcli) not found
creating build
creating build/lib
creating build/lib/loadimpactcli
copying loadimpactcli/__init__.py -> build/lib/loadimpactcli
copying loadimpactcli/client.py -> build/lib/loadimpactcli
copying loadimpactcli/config.py -> build/lib/loadimpactcli
copying loadimpactcli/datastore_commands.py -> build/lib/loadimpactcli
copying loadimpactcli/errors.py -> build/lib/loadimpactcli
copying loadimpactcli/loadimpact_cli.py -> build/lib/loadimpactcli
copying loadimpactcli/organization_commands.py -> build/lib/loadimpactcli
copying loadimpactcli/userscenario_commands.py -> build/lib/loadimpactcli
copying loadimpactcli/version.py -> build/lib/loadimpactcli
file loadimpactcli.py (for module loadimpactcli) not found
creating build/bdist.macosx-10.11-intel
creating build/bdist.macosx-10.11-intel/egg
creating build/bdist.macosx-10.11-intel/egg/loadimpactcli
copying build/lib/loadimpactcli/__init__.py -> build/bdist.macosx-10.11-intel/egg/loadimpactcli
copying build/lib/loadimpactcli/client.py -> build/bdist.macosx-10.11-intel/egg/loadimpactcli
copying build/lib/loadimpactcli/config.py -> build/bdist.macosx-10.11-intel/egg/loadimpactcli
copying build/lib/loadimpactcli/datastore_commands.py -> build/bdist.macosx-10.11-intel/egg/loadimpactcli
copying build/lib/loadimpactcli/errors.py -> build/bdist.macosx-10.11-intel/egg/loadimpactcli
copying build/lib/loadimpactcli/loadimpact_cli.py -> build/bdist.macosx-10.11-intel/egg/loadimpactcli
copying build/lib/loadimpactcli/organization_commands.py -> build/bdist.macosx-10.11-intel/egg/loadimpactcli
copying build/lib/loadimpactcli/userscenario_commands.py -> build/bdist.macosx-10.11-intel/egg/loadimpactcli
copying build/lib/loadimpactcli/version.py -> build/bdist.macosx-10.11-intel/egg/loadimpactcli
byte-compiling build/bdist.macosx-10.11-intel/egg/loadimpactcli/__init__.py to __init__.pyc
byte-compiling build/bdist.macosx-10.11-intel/egg/loadimpactcli/client.py to client.pyc
byte-compiling build/bdist.macosx-10.11-intel/egg/loadimpactcli/config.py to config.pyc
byte-compiling build/bdist.macosx-10.11-intel/egg/loadimpactcli/datastore_commands.py to datastore_commands.pyc
byte-compiling build/bdist.macosx-10.11-intel/egg/loadimpactcli/errors.py to errors.pyc
byte-compiling build/bdist.macosx-10.11-intel/egg/loadimpactcli/loadimpact_cli.py to loadimpact_cli.pyc
byte-compiling build/bdist.macosx-10.11-intel/egg/loadimpactcli/organization_commands.py to organization_commands.pyc
byte-compiling build/bdist.macosx-10.11-intel/egg/loadimpactcli/userscenario_commands.py to userscenario_commands.pyc
byte-compiling build/bdist.macosx-10.11-intel/egg/loadimpactcli/version.py to version.pyc
installing package data to build/bdist.macosx-10.11-intel/egg
running install_data
copying README.md -> build/bdist.macosx-10.11-intel/egg/
creating build/bdist.macosx-10.11-intel/egg/EGG-INFO
copying loadimpact_cli.egg-info/PKG-INFO -> build/bdist.macosx-10.11-intel/egg/EGG-INFO
copying loadimpact_cli.egg-info/SOURCES.txt -> build/bdist.macosx-10.11-intel/egg/EGG-INFO
copying loadimpact_cli.egg-info/dependency_links.txt -> build/bdist.macosx-10.11-intel/egg/EGG-INFO
copying loadimpact_cli.egg-info/entry_points.txt -> build/bdist.macosx-10.11-intel/egg/EGG-INFO
copying loadimpact_cli.egg-info/requires.txt -> build/bdist.macosx-10.11-intel/egg/EGG-INFO
copying loadimpact_cli.egg-info/top_level.txt -> build/bdist.macosx-10.11-intel/egg/EGG-INFO
zip_safe flag not set; analyzing archive contents...
creating dist
creating 'dist/loadimpact_cli-1.0.6-py2.7.egg' and adding 'build/bdist.macosx-10.11-intel/egg' to it
removing 'build/bdist.macosx-10.11-intel/egg' (and everything under it)
Processing loadimpact_cli-1.0.6-py2.7.egg
Copying loadimpact_cli-1.0.6-py2.7.egg to /Library/Python/2.7/site-packages
Adding loadimpact-cli 1.0.6 to easy-install.pth file
Installing loadimpact script to /usr/local/bin

Installed /Library/Python/2.7/site-packages/loadimpact_cli-1.0.6-py2.7.egg
Processing dependencies for loadimpact-cli==1.0.6
Searching for mock
Reading https://pypi.python.org/simple/mock/
Best match: mock 2.0.0
Downloading https://pypi.python.org/packages/0c/53/014354fc93c591ccc4abff12c473ad565a2eb24dcd82490fae33dbf2539f/mock-2.0.0.tar.gz#md5=0febfafd14330c9dcaa40de2d82d40ad
Processing mock-2.0.0.tar.gz
Writing /tmp/easy_install-2hFZun/mock-2.0.0/setup.cfg
Running mock-2.0.0/setup.py -q bdist_egg --dist-dir /tmp/easy_install-2hFZun/mock-2.0.0/egg-dist-tmp-Nq2iHD
error: Installed distribution setuptools 1.1.6 conflicts with requirement setuptools>=17.1
```

4. Seems installation was partly successful, but something went wrong at the end. I do have a binary in /usr/local, trying to run that:

```
Ragnars-MacBook-Air:loadimpact-cli ragnarlonn$ /usr/local/bin/loadimpact 
Traceback (most recent call last):
  File "/usr/local/bin/loadimpact", line 5, in <module>
    from pkg_resources import load_entry_point
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/pkg_resources.py", line 2793, in <module>
    working_set.require(__requires__)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/pkg_resources.py", line 673, in require
    needed = self.resolve(parse_requirements(requirements))
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/pkg_resources.py", line 576, in resolve
    raise DistributionNotFound(req)
pkg_resources.DistributionNotFound: mock
```

No luck. It's definitely not plug-and-play for me, this app!  After upgrading pip I still get the same error when trying to do 'sudo pip install loadimpact-cli', so the old pip version doesn't seem to have anything to do with it.

**Comment by ragnarlonn** [comment]
Tried installing it on an Ubuntu 14.04 server also (snik1):

```
root@snik1:/home/ragnar# pip install loadimpact-cli
Downloading/unpacking loadimpact-cli
  Downloading loadimpact-cli-1.0.6.tar.gz
  Running setup.py (path:/tmp/pip_build_root/loadimpact-cli/setup.py) egg_info for package loadimpact-cli
    /usr/lib/python2.7/distutils/dist.py:267: UserWarning: Unknown distribution option: 'test_requires'
      warnings.warn(msg)
    
    file loadimpactcli.py (for module loadimpactcli) not found
    warning: no previously-included files matching '__pycache__' found anywhere in distribution
Downloading/unpacking setuptools>=18 (from loadimpact-cli)
  Downloading setuptools-34.1.0-py2.py3-none-any.whl (389kB): 389kB downloaded
Requirement already satisfied (use --upgrade to upgrade): click in /usr/local/lib/python2.7/dist-packages (from loadimpact-cli)
Downloading/unpacking loadimpact-v3 (from loadimpact-cli)
  Downloading loadimpact-v3-0.0.3.tar.gz
  Running setup.py (path:/tmp/pip_build_root/loadimpact-v3/setup.py) egg_info for package loadimpact-v3
    /usr/lib/python2.7/distutils/dist.py:267: UserWarning: Unknown distribution option: 'test_requires'
      warnings.warn(msg)
    
Downloading/unpacking tzlocal (from loadimpact-cli)
  Downloading tzlocal-1.3.tar.gz
  Running setup.py (path:/tmp/pip_build_root/tzlocal/setup.py) egg_info for package tzlocal
    
Requirement already satisfied (use --upgrade to upgrade): six in /usr/lib/python2.7/dist-packages (from loadimpact-cli)
Downloading/unpacking mock (from loadimpact-cli)
  Downloading mock-2.0.0-py2.py3-none-any.whl (56kB): 56kB downloaded
Downloading/unpacking appdirs>=1.4.0 (from setuptools>=18->loadimpact-cli)
  Downloading appdirs-1.4.0-py2.py3-none-any.whl
Downloading/unpacking packaging>=16.8 (from setuptools>=18->loadimpact-cli)
  Downloading packaging-16.8-py2.py3-none-any.whl
Requirement already satisfied (use --upgrade to upgrade): requests in /usr/local/lib/python2.7/dist-packages (from loadimpact-v3->loadimpact-cli)
Requirement already satisfied (use --upgrade to upgrade): pytz in /usr/lib/python2.7/dist-packages (from tzlocal->loadimpact-cli)
Downloading/unpacking funcsigs>=1 (from mock->loadimpact-cli)
  Downloading funcsigs-1.0.2-py2.py3-none-any.whl
Downloading/unpacking pbr>=0.11 (from mock->loadimpact-cli)
  Downloading pbr-1.10.0-py2.py3-none-any.whl (96kB): 96kB downloaded
Requirement already satisfied (use --upgrade to upgrade): pyparsing in /usr/lib/python2.7/dist-packages (from packaging>=16.8->setuptools>=18->loadimpact-cli)
Installing collected packages: loadimpact-cli, setuptools, loadimpact-v3, tzlocal, mock, appdirs, packaging, funcsigs, pbr
  Running setup.py install for loadimpact-cli
    /usr/lib/python2.7/distutils/dist.py:267: UserWarning: Unknown distribution option: 'test_requires'
      warnings.warn(msg)
    file loadimpactcli.py (for module loadimpactcli) not found
    
    file loadimpactcli.py (for module loadimpactcli) not found
    warning: no previously-included files matching '__pycache__' found anywhere in distribution
    file loadimpactcli.py (for module loadimpactcli) not found
    Installing loadimpact script to /usr/local/bin
    file loadimpactcli.py (for module loadimpactcli) not found
  Found existing installation: setuptools 3.3
    Not uninstalling setuptools at /usr/lib/python2.7/dist-packages, owned by OS
  Running setup.py install for loadimpact-v3
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/usr/local/lib/python2.7/dist-packages/setuptools/__init__.py", line 12, in <module>
        import setuptools.version
      File "/usr/local/lib/python2.7/dist-packages/setuptools/version.py", line 1, in <module>
        import pkg_resources
      File "/usr/local/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 70, in <module>
        import packaging.version
    ImportError: No module named packaging.version
    Complete output from command /usr/bin/python -c "import setuptools, tokenize;__file__='/tmp/pip_build_root/loadimpact-v3/setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record /tmp/pip-FoQdjV-record/install-record.txt --single-version-externally-managed --compile:
    Traceback (most recent call last):

  File "<string>", line 1, in <module>

  File "/usr/local/lib/python2.7/dist-packages/setuptools/__init__.py", line 12, in <module>

    import setuptools.version

  File "/usr/local/lib/python2.7/dist-packages/setuptools/version.py", line 1, in <module>

    import pkg_resources

  File "/usr/local/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 70, in <module>

    import packaging.version

ImportError: No module named packaging.version

----------------------------------------
Cleaning up...
Command /usr/bin/python -c "import setuptools, tokenize;__file__='/tmp/pip_build_root/loadimpact-v3/setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record /tmp/pip-FoQdjV-record/install-record.txt --single-version-externally-managed --compile failed with error code 1 in /tmp/pip_build_root/loadimpact-v3
Storing debug log for failure in /root/.pip/pip.log
```

**Comment by dbrandt** [comment]
Sorry you're having problems. I'll take a look at this!

**Comment by dbrandt** [comment]
A bunch of things to note here. Sorry for the long reply, hopefully it's useful for others as well.

## OS X
The first problem you encounter is helped by `sudo` just like you write. The errors that follow after that seems to be tied to Apple's [System Integrity Protection](https://support.apple.com/en-us/HT204899) which is a fancy way to say they mounted some directories read-only to prevent tampering by malware and such. Unfortunately, some of the installed components `pip` wants to touch are located in `/System` which is protected this way.

The problem that arise when you try to run `python setup.py install` manually seems to be due to a super-outdated version of setuptools.

All in all, these problems are symptomatic of the Apple supplied Python environment. Your two best options at this stage are:
* Use MacPorts or Homebrew (my personal choice) and install the Python versions you need and keep them up to date that way. These package managers usually keep their libraries in `/usr/local` which is excluded from the System Integrity Protection. Install all external modules with [`pip` installed manually](https://pip.pypa.io/en/stable/installing/) to ensure it's using the right python binary.
* Use the shipped Python interpreter, but install `virtualenv` and install what you need that way.

## Ubuntu 14.04
I get the same error when using `pip` as installed by `apt` on a clean image of ubuntu 14.04. As a matter of fact, `pip` broke after I tried to install `loadimpact-cli`. This is likely due to a `setuptools`/`distribute`-clash. Several projects have come and gone that's promised to clean up Python packaging, and this is a consequence of that. I've seen similar things happen with other packages as well. My prefered way out of this mess is installing `pip` manually. In fact, not installing Python packages via the system package manager is almost always a better choice. To install `pip` manually follow the [directions](https://pip.pypa.io/en/stable/installing/).

After this, your `pip` install should work (even for the `loadimpact-cli` package), but will keep warning about insecure use of SSL, which unfortunately is a consequence of the version of Python 2.7 that's shipped with ubuntu 14.04.

**Comment by dbrandt** [comment]
I added a note about using `virtualenv` to avoid this problem to the README.
